### PR TITLE
Add [auth|security] checkRights

### PR DESCRIPTION
**Pull request by y-abs**
## What does this PR do ?

Add auth/security checkRights API action https://github.com/kuzzleio/sdk-dart/issues/53

### CODE REVIEW

File: doc/2/controllers/auth/check-rights/snippets/check-rights.dart [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  'controller': 'document',
+  'action': 'create',
+  'index': 'nyc-open-data',
+  'collection': 'yellow-taxi',
+  'body': {
+    'name': 'Melis'
+  }
+};
+
+final result = await kuzzle
```
Comment (Aschen) [inline doc/2/controllers/auth/check-rights/snippets/check-rights.dart:11]:
```suggestion
final allowed = await kuzzle
```

File: doc/2/controllers/security/check-rights/snippets/check-rights.dart [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  'controller': 'document',
+  'action': 'create',
+  'index': 'nyc-open-data',
+  'collection': 'yellow-taxi',
+  'body': {
+    'name': 'Melis'
+  }
+};
+
+final result = await kuzzle
```
Comment (Aschen) [inline doc/2/controllers/security/check-rights/snippets/check-rights.dart:11]:
```suggestion
final allowed = await kuzzle
```

**Review by Aschen — APPROVED** [review]

File: doc/2/controllers/auth/check-rights/snippets/check-rights.dart (hunk 1 above)
Comment (y-abs) [reply doc/2/controllers/auth/check-rights/snippets/check-rights.dart:11]: It must be named `result` as the template is printing it.

File: doc/2/controllers/auth/check-rights/index.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+---
+code: true
+type: page
+title: checkRights
+description: Checks if the provided API request can be executed by this network connection, using the current authentication information.
```
Comment (xbill82) [inline doc/2/controllers/auth/check-rights/index.md]:
```suggestion
description: Checks if the provided API request can be executed by this client, using the current authentication information.
```

File: doc/2/controllers/auth/check-rights/index.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+---
+
+# checkRights
+
+<SinceBadge version="Kuzzle 2.8.0"/>
+<SinceBadge version="auto-version"/>
+
+Checks if the provided API request can be executed by this network connection, using the current authentication information.
```
Comment (xbill82) [inline doc/2/controllers/auth/check-rights/index.md]:
```suggestion
Checks if the provided API request can be executed by this client, using the current authentication information.
```

File: doc/2/controllers/auth/check-rights/index.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    Map<String, dynamic> requestPayload)
+```
+
+| Property | Type | Description |
+|--- |--- |--- |
+| `requestPayload` | <pre>Map<String, dynamic></pre> | Contains a [RequestPayload](/core/2/api/payloads/request) |
+
+## `requestPayload`
+
+The [RequestPayload](/core/2/api/payloads/request) must contains at least the following properties:
```
Comment (xbill82) [inline doc/2/controllers/auth/check-rights/index.md]:
```suggestion
The [RequestPayload](/core/2/api/payloads/request) must contain at least the following properties:
```

File: doc/2/controllers/auth/check-rights/index.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+- `controller`: API controller
+- `action`: API action
+
+---
+
+## Returns
+
+Returns a boolean telling whether the provided request would have been allowed or not
```
Comment (xbill82) [inline doc/2/controllers/auth/check-rights/index.md]:
```suggestion
Returns a boolean telling whether the provided request is allowed or not.
```

File: doc/2/controllers/security/check-rights/index.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+type: page
+title: checkRights
+description: Checks if an API action can be executed by the current user
+---
+
+# checkRights
+
+<SinceBadge version="2.8.0"/>
+
+Checks if the provided API request can be executed by a user.
```
Comment (xbill82) [inline doc/2/controllers/security/check-rights/index.md]:
```suggestion
Checks if the provided API request can be executed by the current user.
```

File: doc/2/controllers/security/check-rights/index.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+```
+
+| Property | Type | Description |
+|--- |--- |--- |
+| `kuid` | <pre>String</pre> | User [kuid](/core/2/guides/main-concepts/authentication#kuzzle-user-identifier-kuid) |
+| `requestPayload` | <pre>Map<String, dynamic></pre> | Contains a [RequestPayload](/core/2/api/payloads/request) |
+
+## `requestPayload`
+
+The [RequestPayload](/core/2/api/payloads/request) must contains at least the following properties:
```
Comment (xbill82) [inline doc/2/controllers/security/check-rights/index.md]:
```suggestion
The [RequestPayload](/core/2/api/payloads/request) must contain at least the following properties:
```

File: doc/2/controllers/security/check-rights/index.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+- `controller`: API controller
+- `action`: API action
+
+---
+
+## Returns
+
+Returns a boolean telling whether the provided request would have been allowed or not
```
Comment (xbill82) [inline doc/2/controllers/security/check-rights/index.md]:
```suggestion
Returns a boolean telling whether the provided request is allowed or not.
```

**Review by xbill82 — APPROVED** [review]
Typos and nitpicking

**Pull request merged**